src/constructors/styled.js

```diff
@@ -1,14 +1,14 @@
 // @flow
 import type { Target } from '../types'
-import domElements from '../utils/domElements'
 
 export default (styledComponent: Function, constructWithOptions: Function) => {
-  const styled = (tag: Target) => constructWithOptions(styledComponent, tag)
+  const baseStyled = (tag: Target) => constructWithOptions(styledComponent, tag)
 
-  // Shorthands for all valid HTML Elements
-  domElements.forEach(domElement => {
-    styled[domElement] = styled(domElement)
-  })
+  const shorthand = {
+    get: (target, name) => (name in target ? target[name] : baseStyled(name)),
+  }
+
+  const styled = new Proxy(baseStyled, shorthand)
```

Comment:
Unfortunately we can't do this I think, we have to support IE9+ which don't have `Proxy`?

Reply:
nope, we cannot do this. We discussed it before and there's no way to get a polyfill for it. For quite obvious reasons we cannot get rid of the known list and at the same time polyfill Proxies.